lib/MusicBrainz/Server/Entity/Medium.pm

```diff
@@ -96,40 +97,26 @@ sub may_have_discids {
 
 =head2 length
 
-Attempt to return the duration of the medium in microseconds.  This
-will return the length of the disc, by looking at the associated
-discids or tracklists.
-
-This will not load any data from the database, so make sure you load
-either the associated tracklists + tracks, the MediumCDTOC +
-CDTOC, or both.
+Returns the duration of the medium in microseconds, including
+possible pregap and data tracks.
+If we have some tracks for which we are missing the durations,
+returns undefined (since we don't actually know the true
+duration of the medium).
 
 =cut
 
 sub length {
     my $self = shift;
 
-    if (scalar $self->all_tracks > 0)
-    {
-        my $length = 0;
-
-        for my $trk ($self->all_tracks)
-        {
-            return undef unless defined $trk->length;
+    my @track_times = (
+        @{ $self->pregap_length // [] },
+        @{ $self->cdtoc_track_lengths // [] },
+        @{ $self->data_track_lengths // [] },
+    );
```

Comment:
This previously used `push` to append the latter two arrays. I changed it to put them all in the declaration, which is equivalent.